### PR TITLE
Add doc for sampling

### LINKED ISSUE #1679: Docs page "Reference Guide > Core Concepts > Sampling"

### PR DESCRIPTION
**Pull request by yeukfu**
Close #1679

### CODE REVIEW

**Comment by ryanjulian** [comment]
@krzentner please review this

@yeukfu Please make sure your doc includes all of  the information I covered in https://github.com/rlworkgroup/garage/issues/1743#issuecomment-656383011

File: docs/user/sampling.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Sampler
+
+Sampler is responsible for assign sampling jobs to workers. Garage now has
+three types of `Sampler`s:
+
+- [`LocalSampler`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/local_sampler.py),
+the default sampler, which runs workers in the main process. With this sampler,
+all the sampling tasks will be run in the same thread.
+
+- [`MultiprocessingSampler`](https://github.com/rlworkgroup/garage/blob/master/src/garage/sampler/multiprocessing_sampler.py),
```
Comment (krzentner) [inline docs/user/sampling.md:19]: You probably shouldn't put `MultiprocessingSampler` here. It currently only works with PyTorch CPU. Once I make it work with more backends I'll add it to this document.

File: docs/user/sampling.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Sampling
+
+`LocalRunner` gets trajectories through Sampling to train the policy. In
+Garage, [`LocalRunner`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/experiment/local_runner.py)
+uses [`Sampler`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/sampler.py)
+to perform sampling. `Sampler` manages [`Worker`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/worker.py#L5)s
+and assign specific tasks to them, such as performing rollouts and update
```
Comment (krzentner) [inline docs/user/sampling.md:7]: This isn't really accurate. All workers basically just do rollouts. However, they may have different environments or agents.

File: docs/user/sampling.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- [`RaySampler`](https://github.com/rlworkgroup/garage/blob/master/src/garage/sampler/ray_sampler.py),
+the sampler use [ray](https://github.com/ray-project/ray) framework to run
+distributed workers. `RaySampler` can not only run workers in different CPUs,
+but also in different machines.
+
+## Worker
+
+Worker is the base unit to perform a rollout and update the agent and the
+environment. In paralleling samplers, a worker will typically run in one
+exclusive CPU. Garage currently have two kinds of workers:
```
Comment (krzentner) [inline docs/user/sampling.md:33]: You should mention that these are the two workers that can be used for most Algorithms, but a few Algorithms (RL2 and PEARL) use custom workers specific to that Algorithm.

File: docs/user/sampling.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```py
+from garage.sampler import RaySampler, VecWorker
+
+...
+runner.setup(
+    algo=algo,
+    env=env,
+    sampler_cls=RaySampler,
+    n_workers=4,
+    worker_class=VecWorker
```
Comment (krzentner) [inline docs/user/sampling.md:60]: Please also show how to set the level of vectorization.

Comment (yeukfu) [reply docs/user/sampling.md:60]: Does the level of vectorization mean `n_envs` of `VecWorker`?

**Comment by yeukfu** [comment]
@krzentner @ryanjulian The doc has been updated according to comments.

File: docs/user/sampling.md (hunk 4 above)
Comment (krzentner) [reply docs/user/sampling.md]: Yes.

File: docs/user/sampling.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+## Sampler
+
+Sampler is responsible for assign sampling jobs to workers. Garage now has
+two types of `Sampler`s:
+
+- [`LocalSampler`](https://github.com/rlworkgroup/garage/blob/a767dcb95988abc14165a2a320ee39932a1c85ca/src/garage/sampler/local_sampler.py),
+the default sampler, which runs workers in the main process in serial style.
+With this sampler, all the sampling tasks will run in the same thread.
+
+- [`RaySampler`](https://github.com/rlworkgroup/garage/blob/master/src/garage/sampler/ray_sampler.py),
```
Comment (ryanjulian) [inline docs/user/sampling.md:20]: What about MultiprocessingSampler?

Comment (yeukfu) [reply docs/user/sampling.md:20]: KR said that he will add it when it works with more backends.

**Pull request merged**